### ISSUE TITLE
AppManager can upgrade a pinned app

### ISSUE THREAD

**Issue by sohkai** [aragon/aragonOS#332]

**Comment by izqui** [comment]
I disagree with the wording of the issue, as the app cannot be upgraded but references to the app can be removed if not protected correctly.

Something that we could do, is allowing to do a `setApp(..)` in a way that the reference cannot be changed. We could then even remove `AppProxyPinned` (though interacting with pinned apps would be more expensive as it would need to `CALL` rather than `SLOAD` to find its implementation code).

Again, this would only hold true if the Kernel code itself is set as an unchangeable reference, otherwise you could always upgrade the Kernel to some code that ignores this premise.

**Comment by sohkai** [comment]
Updated to use "replace" instead of "update", to better describe the issue.

I've been thinking of similar "permanent" storage guarantees, but yes, we should clearly document that any "guarantee" can go away if the Kernel code is changed.
